src/condor_starter.V6.1/os_proc.cpp

```diff
@@ -470,7 +467,7 @@ OsProc::StartJob(FamilyInfo* family_info, FilesystemRemap* fs_remap=NULL)
 		if (param_boolean("SINGULARITY_USE_LAUNCHER", false)) {
 			launcher = htcondor::Singularity::USE_LAUNCHER;
 		}
-		sing_result = htcondor::Singularity::setup(*starter->jic->machClassAd(), *JobAd, JobName, args, job_iwd ? job_iwd : "", execute_dir, job_env, launcher);
+		sing_result = htcondor::Singularity::setup(*starter->jic->machClassAd(), *JobAd, JobName, args, starter->GetSlotDir(), job_iwd ? job_iwd : "", starter->GetWorkingDir(0), job_env, launcher);
```

Comment:
I really like `GetSlotDir()` as a function name, so now I'm sad about `GetWorkingDir(0)`.

I would like to see:
* `Get[Job]ScratchDirectory()`
* `Get[Job]HomeDirectory()`
* `Get[Job]CondorDirectory()` (maybe `GetJobDataDir()`?)

I mention this because it's now unclear why one of these calls is `GetSlotDir()` and the other `GetWorkingDIr(0)`, and we may generally want to replace calls to `GetWorkingDir(0)` with the `GetSlotDir()` -- or whatever -- as appropriate.